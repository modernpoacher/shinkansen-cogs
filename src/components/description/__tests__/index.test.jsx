import React from 'react'
import renderer from 'react-test-renderer'

import Description from '#cogs/components/description'

describe('#cogs/components/description', () => {
  describe('<Description />', () => {
    describe('With required props', () => {
      const component = (
        <Description />
      )

      it('renders', () => {
        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })

      describe('`getClassName`', () => {
        it('is defined', () => {
          return expect(Description.prototype.getClassName)
            .toBeDefined()
        })
      })

      describe('`shouldComponentUpdate`', () => {
        it('is defined', () => {
          return expect(Description.prototype.shouldComponentUpdate)
            .toBeDefined()
        })
      })
    })

    describe('With additional props', () => {
      it('renders', () => {
        const component = (
          <Description
            description='MOCK DESCRIPTION'
          />
        )

        return expect(renderer.create(component).toJSON())
          .toMatchSnapshot()
      })
    })

    describe('`getClassName()`', () => {
      it('returns the classname', () => {
        const component = (
          <Description />
        )

        const instance = (
          renderer.create(component)
            .getInstance()
        )

        return expect(instance.getClassName())
          .toBe('description')
      })
    })

    describe('`shouldComponentUpdate()`', () => {
      const component = (
        <Description
          description='MOCK DESCRIPTION'
        />
      )

      let instance

      beforeEach(() => {
        instance = (
          renderer.create(component)
            .getInstance()
        )
      })

      describe('`props` have changed', () => {
        it('returns true', () => {
          return expect(instance.shouldComponentUpdate({
            description: 'MOCK CHANGE DESCRIPTION'
          }))
            .toBe(true)
        })
      })

      describe('`props` have not changed', () => {
        it('returns false', () => {
          return expect(instance.shouldComponentUpdate({ // instance.props
            description: 'MOCK DESCRIPTION'
          }))
            .toBe(false)
        })
      })
    })
  })
})
