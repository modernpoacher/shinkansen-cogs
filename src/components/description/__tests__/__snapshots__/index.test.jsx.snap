// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#cogs/components/description <Description /> With additional props renders 1`] = `
<span
  className="description"
>
  <span
    className="text-content"
  >
    MOCK DESCRIPTION
  </span>
</span>
`;

exports[`#cogs/components/description <Description /> With required props renders 1`] = `null`;
